src/components/TradeWidget/index.tsx

```diff
@@ -74,17 +86,79 @@ const TradeWidget: React.FC = () => {
     }
   }
 
+  async function placeOrder(
+    buyAmount: BN,
+    buyToken: TokenDetails,
+    sellAmount: BN,
+    sellToken: TokenDetails,
+  ): Promise<void> {
```

Comment:
I really think we should switch to object as parameter for many-parameters functions

Reply:
I can start with this function and create a task in the backlog for refactoring others.

Reply:
Refactored.